### ISSUE TITLE
Observer in Quantization throws Warning

### ISSUE THREAD

**Issue by FabianSchuetze** [pytorch/pytorch#40278]
## To Reproduce
Run the [static quantization tutorial](https://pytorch.org/tutorials/advanced/static_quantization_tutorial.html), in particular the lines:
```python
per_channel_quantized_model = load_model(saved_model_dir + float_model_file)
per_channel_quantized_model.eval()
per_channel_quantized_model.fuse_model()
per_channel_quantized_model.qconfig = torch.quantization.get_default_qconfig('fbgemm')
print(per_channel_quantized_model.qconfig)

torch.quantization.prepare(per_channel_quantized_model, inplace=True)
evaluate(per_channel_quantized_model,criterion, data_loader, num_calibration_batches)
torch.quantization.convert(per_channel_quantized_model, inplace=True)
```
Then, the last line throws the following warning: 
```shell
/home/fabian/.local/lib/python3.7/site-packages/torch/quantization/observer.py:845: UserWarning: must run observer before calling calculate_qparams.                                    Returning default scale and zero point 
  Returning default scale and zero point "
```

## Expected behavior
Going through the source code, I think the warning means that no boundaries for the quantization are calculated but that the function returns instead default boundaries.  If this is not a bug, could somebody kindly illustrate how I can avoid the warning and insert the observers correctly? 

## Environment
Collecting environment information...
PyTorch version: 1.5.0+cu101
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 19.10
GCC version: (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
CMake version: version 3.13.4

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.168
GPU models and configuration: GPU 0: Quadro T1000
Nvidia driver version: 435.21
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.4

Versions of relevant libraries:
[pip3] numpy==1.16.0
[pip3] torch==1.5.0+cu101
[pip3] torchvision==0.6.0+cu101
[conda] Could not collect

**Comment by z-a-f** [comment]
We are aware of this warning, and we also know it sometimes throws a false positive. @raghuramank100 is there a tracked issue for this one?

**Comment by FabianSchuetze** [comment]
Thanks for your reply, @z-a-f ! 

1. As you say it might be a false positive: Do I have any way to determine independently whether the observed did indeed collect information? 
2. Do you know whether this issue might not occur in previous version of pytorch such as 1.3 ? I originally tried to use quantization for an SSD-type model which leads to low accuracy after quantization and after throwing the same warning. I wonder if it might make sense to run the code with an earlier version of pytorch.

**Comment by khizar-anjum** [comment]
Hi, 

Chiming in to confirm that the issue persists on Pytorch 1.3.1. Here is my system information:

PyTorch version: 1.3.1+cu92
Is debug build: No
CUDA used to build PyTorch: 9.2

OS: Ubuntu 20.02
GCC version: gcc (Ubuntu 7.5.0-6ubuntu2) 7.5.0

Python version: 3.6.9
Is CUDA available: Yes
CUDA runtime version: 9.2.148
GPU models and configuration: GPU 0: GTX 1070 Ti
Nvidia driver version: 440.64
cuDNN version: 7.6.5

Versions of relevant libraries:
[pip3] numpy==1.17.4
[pip3] torch==1.3.1+cu92
[pip3] torchvision==0.4.2+cu92

**Comment by FabianSchuetze** [comment]
Thanks a lot for the info, @khizar-anjum !

**Comment by FabianSchuetze** [comment]
I have the impression that the error might be caused by a spurious network configuration containing functions that are never used and for which indeed no observations exists.

In particular, in the [InvertedResidual](https://pytorch.org/tutorials/advanced/static_quantization_tutorial.html) class of the tutorial a `skip_add` module is inserted. However, this `skip_add` module is used only conditionally (depending on `self.use_res_connect`) which means that the observer might indeed not collect information. When I avoid creating the `skip_add` module and instead simply add  `x + self.conv(x)` no spurious `skip_add` modules are created and the `UserWarning` disappears.

**Comment by vkuzo** [comment]
if this reproes on the tutorial, we should call it out in the tutorial

**Comment by EchoIR** [comment]
> Thanks for your reply, @z-a-f !
> 
> 1. As you say it might be a false positive: Do I have any way to determine independently whether the observed did indeed collect information?
> 2. Do you know whether this issue might not occur in previous version of pytorch such as 1.3 ? I originally tried to use quantization for an SSD-type model which leads to low accuracy after quantization and after throwing the same warning. I wonder if it might make sense to run the code with an earlier version of pytorch.

您好，请问，ssd的量化精度提升了嘛，我量化之后差了10多个点

**Comment by HDCharles** [comment]
yeah, not all parts of the model are run for each model run, parts that skipped still throw the warning, this is expected, added a warning to the tutorial about this.
